include/options/option/abstract-option.php

```diff
@@ -0,0 +1,316 @@
+<?php
+/**
+ * @package Polylang
+ */
+
+namespace WP_Syntex\Polylang\Options\Option;
+
+use WP_Error;
+use WP_Syntex\Polylang\Options\Options;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class defining a single option.
+ *
+ * @since 3.7
+ *
+ * @phpstan-type SchemaType 'string'|'null'|'number'|'integer'|'boolean'|'array'|'object'
+ * @phpstan-type Schema array{
+ *     '$schema': non-falsy-string,
+ *     title: non-falsy-string,
+ *     description: string,
+ *     type: SchemaType,
+ *     context: array<non-falsy-string>
+ * }&array<non-falsy-string, mixed>
+ */
+abstract class Abstract_Option {
+	/**
+	 * Option key.
+	 *
+	 * @var string
+	 * @phpstan-var non-falsy-string
+	 */
+	private $key;
+
+	/**
+	 * Option value.
+	 *
+	 * @var mixed
+	 */
+	private $value;
+
+	/**
+	 * Option default value.
+	 *
+	 * @var mixed
+	 */
+	private $default;
+
+	/**
+	 * Cached option JSON schema.
+	 *
+	 * @var array|null
+	 */
+	private $schema;
```

Comment:
`null` is used [at the beginning of `get_schema()`](https://github.com/polylang/polylang/pull/1451/files#diff-9f339aa049cf454580a6354b6bbdc9da9d89cc7b5ad36e8bf50eaf78f3f9f4f0R174). Do you prefer we keep `array` only and then check for `! empty()` instead of `is_array()`?

Reply:
As we discussed, we'll keep it like this.